core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -2924,6 +2919,15 @@ private QueryModel rewriteOrderBy(QueryModel model) throws SqlException {
 
         while (base.getBottomUpColumns().size() > 0 && !base.isNestedModelIsSubQuery()) {
             baseParent = base;
+
+            final QueryModel union = base.getUnionModel();
+            if (union != null) {
+                final QueryModel rewritten = rewriteOrderBy(union);
+                if (rewritten != union) {
```

Comment:
Line 2927, like existing line 3064,  is not covered by tests. 
I couldn't figure out SQL to trigger it but would rather keep it for consistency .